modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -0,0 +1,671 @@
+		$jetpack_slug = 'https://wordpress.com/activity-log/' . $domain;
+
+		$this->add_admin_menu_separator( $position++, 'manage_options' );
+		add_menu_page( esc_attr__( 'Jetpack', 'jetpack' ), __( 'Jetpack', 'jetpack' ), 'manage_options', $jetpack_slug, null, $jetpack_icon, $position );
+
+		// Maintain id for jQuery selector.
+		$menu[ $position ][5] = 'toplevel_page_jetpack'; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		remove_menu_page( 'jetpack' );
+		remove_submenu_page( 'jetpack', esc_url( Redirect::get_url( 'calypso-backups' ) ) );
+
+		$this->migrate_submenus( 'jetpack', $jetpack_slug );
+
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Activity Log', 'jetpack' ), __( 'Activity Log', 'jetpack' ), 'manage_options', 'https://wordpress.com/activity-log/' . $domain, null, 5 );
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Backup', 'jetpack' ), __( 'Backup', 'jetpack' ), 'manage_options', 'https://wordpress.com/backup/' . $domain, null, 10 );
+		add_submenu_page( $jetpack_slug, esc_attr__( 'Scan', 'jetpack' ), __( 'Scan', 'jetpack' ), 'manage_options', 'https://wordpress.com/scan/' . $domain, null, 15 );
+
+		add_filter( 'parent_file', array( $this, 'jetpack_parent_file' ) );
+	}
+
+	/**
+	 * Filters the parent file of an admin menu sub-menu item.
+	 *
+	 * @param string $parent_file The parent file.
+	 * @return string Updated parent file.
+	 */
+	public function jetpack_parent_file( $parent_file ) {
+		if ( 'jetpack' === $parent_file ) {
+			$parent_file = 'https://wordpress.com/activity-log/' . wp_parse_url( get_home_url(), PHP_URL_HOST );
+
+			// TODO: Remove once feature has shipped. See reregister_menu_items().
+			if ( ! $this->is_api_request && ! defined( 'PHPUNIT_JETPACK_TESTSUITE' ) ) {
+				$parent_file = add_query_arg( 'flags', 'nav-unification', $parent_file );
+			}
+		}
+
+		return $parent_file;
+	}
+
+	/**
+	 * Adds Appearance menu.
+	 *
+	 * @param string $domain  Site domain.
+	 * @param bool   $calypso Optional. Whether links should point to Calypso or wp-admin. Default true (Calypso).
+	 */
+	public function add_appearance_menu( $domain, $calypso = true ) {
+		$user_can_customize = current_user_can( 'customize' );
+		$appearance_cap     = $user_can_customize ? 'customize' : 'edit_theme_options';
+		$customize_slug     = $calypso ? 'https://wordpress.com/customize/' . $domain : 'customize.php';
+		$themes_slug        = $calypso ? 'https://wordpress.com/themes/' . $domain : 'themes.php';
+		$appearance_slug    = $user_can_customize ? $customize_slug : $themes_slug;
+		$customize_url      = add_query_arg( 'return', urlencode( remove_query_arg( wp_removable_query_args(), wp_unslash( $_SERVER['REQUEST_URI'] ) ) ), 'customize.php' ); // phpcs:ignore
+
+		remove_menu_page( 'themes.php' );
+		remove_submenu_page( 'themes.php', 'themes.php' );
+		remove_submenu_page( 'themes.php', 'theme-editor.php' );
```

Comment:
Is it problematic that folks will lose access to this option? I know we discourage its use in general, but it remains available in wp-admin anyway, even on Atomic. I wonder if this would cause some issues for folks in support in some cases.

Reply:
@davemart-in Do you have thoughts on the removal/non-removal of theme & plugin editor links?

Reply:
Let me phone a few friends. @kristenzuck, @nerdysandy thoughts on removing links to the theme & plugin editor from a happiness perspective? These pages would still be available if you knew how to get to them, but I've proposed removing them from the default menu. I'm happy to change my position on this one. Just curious if you see these pages used by many customers.